File: Solution in CPP/Que11.cpp
// Merge two sorted arrays 

#include<bits/stdc++.h>
using namespace std;

int main(){
    vector<int>sorted_array1 = {1,2,3,4,5};
    vector<int>sorted_array2 = {2,4,6,9};

    vector<int> ans;
    int ind1 = 0, ind2 = 0;
    while(ind1 < sorted_array1.size() && ind2 < sorted_array2.size()){
        if(sorted_array1[ind1] < sorted_array2[ind2]){
            ans.push_back(sorted_array1[ind1]);
            ind1++;
        }

        else{
            ans.push_back(sorted_array2[ind2]);
            ind2++;
        }
    }

    while(ind1 < sorted_array1.size()){
        ans.push_back(sorted_array1[ind1]);
        ind1 ++;
    }

    while(ind2 < sorted_array2.size()){
        ans.push_back(sorted_array2[ind2]);
        ind2 ++;
    }



    // printing the answer;
    for(int i= 0; i<ans.size(); i++){
        cout<<ans[i]<<" ";
    }

    return 0;
}

File: Solution in RUST/Que11.rs
// Merge two sorted arrays in Rust


fn main() {
    let sorted_array1 = vec![1, 2, 3, 4, 5];
    let sorted_array2 = vec![2, 4, 6, 9];

    let mut ans = Vec::new();
    let (mut ind1, mut ind2) = (0, 0);
    while ind1 < sorted_array1.len() && ind2 < sorted_array2.len() {
        if sorted_array1[ind1] < sorted_array2[ind2] {
            ans.push(sorted_array1[ind1]);
            ind1 += 1;
        } else {
            ans.push(sorted_array2[ind2]);
            ind2 += 1;
        }
    }

    while ind1 < sorted_array1.len() {
        ans.push(sorted_array1[ind1]);
        ind1 += 1;
    }

    while ind2 < sorted_array2.len() {
        ans.push(sorted_array2[ind2]);
        ind2 += 1;
    }

    // Printing the answer
    for num in &ans {
        print!("{} ", num);
    }
}
